File: src/utils/json/serialize/JsonConstants.java
package utils.json.serialize;

/**
 * Json constants
 * 
 * @author Tony Tsang
 *
 */
public class JsonConstants {
    private JsonConstants() {}
    
    public static final int
        TYPE_NULL          = 0x01,
        TYPE_REF           = 0x02,
        TYPE_BOOLEAN_TRUE  = 0x03,
        TYPE_BOOLEAN_FALSE = 0x04,
        TYPE_INT8          = 0x05,
        TYPE_INT16         = 0x06,
        TYPE_INT24         = 0x07,
        TYPE_INT32         = 0x08,
        TYPE_INT40         = 0x09,
        TYPE_INT48         = 0x0A,
        TYPE_INT56         = 0x0B,
        TYPE_INT64         = 0x0C,
        TYPE_N_INT8        = 0x15,
        TYPE_N_INT16       = 0x16,
        TYPE_N_INT24       = 0x17,
        TYPE_N_INT32       = 0x18,
        TYPE_N_INT40       = 0x19,
        TYPE_N_INT48       = 0x1A,
        TYPE_N_INT56       = 0x1B,
        TYPE_N_INT64       = 0x1C,
        TYPE_BIGINTEGER    = 0x20,
        TYPE_BIGDECIMAL    = 0x21,
        TYPE_DATE          = 0x22,
        TYPE_SINGLE        = 0x0D,
        TYPE_DOUBLE        = 0x0E,
        TYPE_SINGLE_ZERO   = 0x1D,
        TYPE_DOUBLE_ZERO   = 0x1E,
        TYPE_STRING        = 0x0F,
        TYPE_ARRAY         = 0x10,
        TYPE_OBJECT        = 0x11,
        TYPE_BINARY        = 0x55;
}

File: src/main/kotlin/utils/json/serialize/JsonConstants.kt
package utils.json.serialize

/**
 * Json constants

 * @author Tony Tsang
 */
object JsonConstants {
    @JvmField
    val TYPE_NULL: Byte = 0x01
    @JvmField
    val TYPE_REF: Byte = 0x02
    @JvmField
    val TYPE_BOOLEAN_TRUE: Byte = 0x03
    @JvmField
    val TYPE_BOOLEAN_FALSE: Byte = 0x04
    @JvmField
    val TYPE_INT8: Byte = 0x05
    @JvmField
    val TYPE_INT16: Byte = 0x06
    @JvmField
    val TYPE_INT24: Byte = 0x07
    @JvmField
    val TYPE_INT32: Byte = 0x08
    @JvmField
    val TYPE_INT40: Byte = 0x09
    @JvmField
    val TYPE_INT48: Byte = 0x0A
    @JvmField
    val TYPE_INT56: Byte = 0x0B
    @JvmField
    val TYPE_INT64: Byte = 0x0C
    @JvmField
    val TYPE_N_INT8: Byte = 0x15
    @JvmField
    val TYPE_N_INT16: Byte = 0x16
    @JvmField
    val TYPE_N_INT24: Byte = 0x17
    @JvmField
    val TYPE_N_INT32: Byte = 0x18
    @JvmField
    val TYPE_N_INT40: Byte = 0x19
    @JvmField
    val TYPE_N_INT48: Byte = 0x1A
    @JvmField
    val TYPE_N_INT56: Byte = 0x1B
    @JvmField
    val TYPE_N_INT64: Byte = 0x1C
    @JvmField
    val TYPE_BIGINTEGER: Byte = 0x20
    @JvmField
    val TYPE_BIGDECIMAL: Byte = 0x21
    @JvmField
    val TYPE_DATE: Byte = 0x22
    @JvmField
    val TYPE_SINGLE: Byte = 0x0D
    @JvmField
    val TYPE_DOUBLE: Byte = 0x0E
    @JvmField
    val TYPE_SINGLE_ZERO: Byte = 0x1D
    @JvmField
    val TYPE_DOUBLE_ZERO: Byte = 0x1E
    @JvmField
    val TYPE_STRING: Byte = 0x0F
    @JvmField
    val TYPE_ARRAY: Byte = 0x10
    @JvmField
    val TYPE_OBJECT: Byte = 0x11
    @JvmField
    val TYPE_BINARY: Byte = 0x55
}
